src/libostree/ostree-repo-commit.c

```diff
@@ -4279,6 +4281,48 @@ ostree_repo_commit_modifier_set_sepolicy (OstreeRepoCommitModifier
   modifier->sepolicy = sepolicy ? g_object_ref (sepolicy) : NULL;
 }
 
+gboolean 
+ostree_repo_commit_modifier_sepolicy_from (OstreeRepoCommitModifier              *modifier,
+                                           OstreeRepo                            *repo,
+                                           const char                            *rev,
+                                           GCancellable                          *cancellable,
+                                           GError                               **error)
+{
+  GLNX_AUTO_PREFIX_ERROR ("setting sepolicy from commit", error);
+  g_autofree char *commit = NULL;
+  g_autoptr(GFile) root = NULL;
+  if (!ostree_repo_read_commit (repo, rev, &root, &commit, cancellable, error))
+    return FALSE;
+  const char policypath[] = "usr/etc/selinux";
+  g_autoptr(GFile) policyroot = g_file_get_child (root, policypath);
+  if (!g_file_query_exists (policyroot, NULL))
+    return TRUE;  /* No policy, nothing to do */
```

Comment:
Hmm, I'm wondering if this should be a hard error instead. It feels like if you call this API, you expect an actual `OstreeSePolicy` to get set on the modifier by the end. For example, making it a `G_IO_ERROR_NOT_FOUND` allows callers to handle it and specify a fallback policy via `set_sepolicy`.

Could also make it a e.g. `allow_noent` parameter.

Reply:
I don't want all build scripts to bifurcate between "support selinux" versus not.  This way e.g. the [dev-overlay script](https://github.com/coreos/coreos-assembler/blob/332c6ab3b91778d904224c3c960d9cc4739d60bd/src/cmd-dev-overlay) wouldn't need to check whether a policy exists...which is what this is doing anyways.

This way we basically just Do The Right Thing - the only reason this isn't on by default is a conservatism around behavioral changes.

Now yes we'd end up with silently unlabeled files if for some reason the policy isn't there - but if the policy isn't there *other* things are going to break too...and unlabeled files will quickly be noticed if the system is enforcing, etc.